### ISSUE TITLE
`aria-haspopup` should have different value than `"true"` for elements other than Menu

### ISSUE THREAD

**Issue by DCzajkowski** [tailwindlabs/headlessui#2099]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

1.7.5

**What browser are you using?**

N/A

**Reproduction URL**

https://headlessui.com/react/listbox

**Describe your issue**

[According to MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup), `aria-haspopup` should be set to `"true"` only if the underlying element that pops up is of role `menu`.

> The presence of the aria-haspopup, with one of six enumerated values - menu, listbox, tree, grid, dialog, or true - indicates the element can trigger a popup and what kind of popup will be displayed. In turn, the element that pops up must be of the role indicated. **The value true is the same as menu.**

Reproduction is [in the docs](https://headlessui.com/react/listbox).

**Expected:**
Listbox uses `role="listbox"`, the button uses `aria-haspopup="listbox"`.

**Actual:**
Listbox uses `role="listbox"`, the button uses `aria-haspopup="true"`.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

This should be fixed by #2101, and will be available in the next release.

You can already try it using:
- `npm install @headlessui/react@insiders`.
- `npm install @headlessui/vue@insiders`.
